File: frontend/services/caseService.ts
import axios from "axios"

const API = axios.create({

  baseURL:
    process.env.NEXT_PUBLIC_API_URL
})

/* =========================================
   FETCH CASES
========================================= */

export async function fetchCases() {

  const response = await API.get(
    "/cases"
  )

  return response.data
}

/* =========================================
   APPROVE CASE
========================================= */

export async function approveCase(
  caseId: string
) {

  const response = await API.post(

    `/cases/${caseId}/approve`
  )

  return response.data
}

/* =========================================
   REJECT CASE
========================================= */

export async function rejectCase(
  caseId: string
) {

  const response = await API.post(

    `/cases/${caseId}/reject`
  )

  return response.data
}

/* =========================================
   FETCH REPORTS
========================================= */

export async function fetchReports() {

  const response = await API.get(
    "/reports"
  )

  return response.data
}

File: backend/app/services/case_service.py
from backend.app.db.database import (
    supabase
)

# ==========================================
# FETCH CASES
# ==========================================

def fetch_cases():

    response = supabase.table(
        "cases"
    ).select("*").execute()

    return response.data

# ==========================================
# INSERT CASE
# ==========================================

def insert_case(case_data: dict):

    response = supabase.table(
        "cases"
    ).insert(case_data).execute()

    return response.data

# ==========================================
# APPROVE CASE
# ==========================================

def approve_case(case_id: str):

    response = supabase.table(
        "cases"
    ).update({

        "status": "APPROVED"

    }).eq(

        "id",
        case_id

    ).execute()

    return response.data

# ==========================================
# REJECT CASE
# ==========================================

def reject_case(case_id: str):

    response = supabase.table(
        "cases"
    ).update({

        "status": "REJECTED"

    }).eq(

        "id",
        case_id

    ).execute()

    return response.data